Mathlib/NumberTheory/LSeries/Dirichlet.lean

```diff
@@ -0,0 +1,415 @@
+  exact mul_mul_mul_comm ..
+
+lemma mul_delta {n : ℕ} (χ : DirichletCharacter ℂ n) : ↗χ * δ = δ :=
+  LSeries.mul_delta <| by rw [cast_one, MulChar.map_one]
+
+lemma delta_mul {n : ℕ} (χ : DirichletCharacter ℂ n) : δ * ↗χ = δ :=
+  mul_comm δ _ ▸ mul_delta ..
+
+open ArithmeticFunction in
+/-- The convolution of a Dirichlet character `χ` with the twist `χ * μ` is `δ`,
+the indicator function of `{1}`. -/
+lemma convolution_mul_moebius {n : ℕ} (χ : DirichletCharacter ℂ n) : ↗χ ⍟ (↗χ * ↗μ) = δ := by
+  have : (1 : ℕ → ℂ) ⍟ (μ ·) = δ := by
+    rw [one_convolution_eq_zeta_convolution, ← one_eq_delta]
+    simp_rw [← natCoe_apply, ← intCoe_apply, coe_mul, coe_zeta_mul_coe_moebius]
+  nth_rewrite 1 [← mul_one ↗χ]
+  simpa only [← mul_convolution_distrib χ 1 ↗μ, this] using mul_delta _
+
+/-- The Dirichlet character mod `0` corresponds to `δ`. -/
+lemma modZero_eq_delta {χ : DirichletCharacter ℂ 0} : ↗χ = δ := by
+  ext n
+  rcases eq_or_ne n 0 with rfl | hn
+  · simp_rw [cast_zero, MulCharClass.map_nonunit χ not_isUnit_zero, delta, if_false]
+  rcases eq_or_ne n 1 with rfl | hn'
+  · simp only [cast_one, map_one, delta, ↓reduceIte]
+  have : ¬ IsUnit (n : ZMod 0) := fun h ↦ hn' <| ZMod.eq_one_of_isUnit_natCast h
+  simp only [MulChar.map_nonunit χ this, delta, hn', ↓reduceIte]
+
+/-- The Dirichlet character mod `1` corresponds to the constant function `1`. -/
+lemma modOne_eq_one {R : Type*} [CommSemiring R] {χ : DirichletCharacter R 1} :
+    ((χ ·) : ℕ → R) = 1 := by
+  ext
+  rw [χ.level_one, MulChar.one_apply (isUnit_of_subsingleton _), Pi.one_apply]
+
+lemma LSeries_modOne_eq : L ↗χ₁ = L 1 :=
+  congr_arg L modOne_eq_one
+
+/-- The L-series of a Dirichlet character mod `N > 0` does not converge absolutely at `s = 1`. -/
+lemma not_LSeriesSummable_at_one {N : ℕ} (hN : N ≠ 0) (χ : DirichletCharacter ℂ N) :
+    ¬ LSeriesSummable ↗χ 1 := by
+  intro h
+  refine (Real.not_summable_indicator_one_div_natCast hN 1) ?_
+  refine h.norm.of_nonneg_of_le (fun m ↦ Set.indicator_apply_nonneg (fun _ ↦ by positivity))
+    (fun n ↦ ?_)
+  -- `Set.indicator {n | ↑n = 1} (fun n ↦ 1 / ↑n) n ≤ ‖term (fun n ↦ χ ↑n) 1 n‖`
+  rw [norm_term_eq, one_re, Real.rpow_one]
+  by_cases hn : n ∈ {(m : ℕ) | (m : ZMod N) = 1}
+  · rw [Set.indicator_of_mem hn]
+    rcases eq_or_ne n 0 with rfl | hn₀
+    · simp only [cast_zero, div_zero, ↓reduceIte, le_refl]
+    simp only [hn₀, ↓reduceIte]
+    gcongr
+    rw [hn, map_one, norm_one]
+  · rw [Set.indicator_of_not_mem hn]
+    positivity
+
+/-- The L-series of a Dirichlet character converges absolutely at `s` if and only if `re s > 1`. -/
+lemma LSeriesSummable_iff {N : ℕ} (hN : N ≠ 0) (χ : DirichletCharacter ℂ N) {s : ℂ} :
+    LSeriesSummable ↗χ s ↔ 1 < s.re := by
+  refine ⟨fun H ↦ ?_, LSeriesSummable_of_bounded_of_one_lt_re fun _ _ ↦ χ.norm_le_one _⟩
+  by_contra! h
+  have h' : s.re ≤ (1 : ℂ).re := by simp only [one_re, h]
+  exact not_LSeriesSummable_at_one hN χ <| LSeriesSummable.of_re_le_re h' H
```

Comment:
golf:
```suggestion
  exact not_LSeriesSummable_at_one hN χ <| LSeriesSummable.of_re_le_re (by simp [h]) H
```

Reply:
`by simp only [one_re, h]` is faster and still fits on the line.